app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala

```diff
@@ -193,46 +192,57 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
 
   def createAccount: Action[AnyContent] = authorisedFromStride { implicit request ⇒
     checkSession(SeeOther(routes.StrideController.getEligibilityPage().url),
-                 whenEligible   = { (eligible, detailsConfirmed, nsiUserInfo) ⇒
+                 whenEligible   = { (eligible, detailsConfirmed, nsiUserInfo, _) ⇒
         if (!detailsConfirmed) {
           SeeOther(routes.StrideController.customerEligible().url)
         } else {
-          helpToSaveConnector.createAccount(CreateAccountRequest(nsiUserInfo, eligible.response.reasonCode, "Stride")).fold(
-            error ⇒ {
-              logger.warn(s"error during create account call, error: $error")
-              SeeOther(routes.StrideController.getErrorPage().url)
-            }, {
-              case AccountCreated ⇒
-                SeeOther(routes.StrideController.getAccountCreatedPage().url)
-              case AccountAlreadyExists ⇒
-                Ok(views.html.account_already_exists())
-            })
+          createAccountAndUpdateSession(nsiUserInfo, eligible, detailsConfirmed, eligible.response.reasonCode, "Stride")
         }
       },
-                 whenIneligible = { (ineligible, nSIUserInfo) ⇒
+                 whenIneligible = { (ineligible, nSIUserInfo, _) ⇒
         if (ineligible.manualCreationAllowed) {
           // send the ineligibility reasonCode to the BE for manual account creation
-          helpToSaveConnector.createAccount(CreateAccountRequest(nSIUserInfo, ineligible.response.reasonCode, "Stride-Manual")).fold(
-            error ⇒ {
-              logger.warn(s"error during create account call, error: $error")
-              SeeOther(routes.StrideController.getErrorPage().url)
-            }, {
-              case AccountCreated ⇒
-                SeeOther(routes.StrideController.getAccountCreatedPage().url)
-              case AccountAlreadyExists ⇒
-                Ok(views.html.account_already_exists())
-            })
+          createAccountAndUpdateSession(nSIUserInfo, ineligible, false, ineligible.response.reasonCode, "Stride-Manual")
         } else {
           SeeOther(routes.StrideController.customerNotEligible().url)
         }
       }
     )
   }(routes.StrideController.createAccount())
 
+  private def createAccountAndUpdateSession(nsiUserInfo:       NSIPayload,
+                                            eligibilityResult: SessionEligibilityCheckResult,
+                                            detailsConfirmed:  Boolean,
+                                            reasonCode:        Int,
+                                            source:            String)(implicit hc: HeaderCarrier, request: Request[_]) = {
+
+      def updateSession(a:                CreateAccountResult,
```

Comment:
this is doing a bit more than `updateSession`. It's currently doing something like `updateSessionAndReturnResult`. What'd make more sense here is to only have this function do the update session function and handle the result of the `for`-comprehension in the `fold` below. It'd make the code better structured and easier to refactor if we ever need to

Reply:
fixed